Core/Src/main.c

```diff
@@ -87,14 +90,40 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  MX_ADC_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  uint8_t START = 1;
+  uint8_t END = 2;
+  uint16_t MAX_ADC = 1023; // 1023 is the max value that can be stored by a 10 bit value
+
+  uint16_t adc_value;
+  uint16_t compare_value;
+  float adc_percent;
+  uint16_t scale_value = 3000; // Scaling the 60000 to the period of 20ms: 60000/20=3000
+
+  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_SET);
+
```

Comment:
Great to bring the CS high to start, just remember to bring it low before communicating with the ADC and high again after (in the while loop).